repos/spack_repo/builtin/packages/cabana_pd/package.py

```diff
@@ -0,0 +1,53 @@
+# Copyright Spack Project Developers. See COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+
+from spack_repo.builtin.build_systems.cmake import CMakePackage
+from spack_repo.builtin.build_systems.cuda import CudaPackage
+from spack_repo.builtin.build_systems.rocm import ROCmPackage
+
+from spack.package import *
+
+
+class CabanaPd(CMakePackage, CudaPackage, ROCmPackage):
+    """Peridynamics with the Cabana library"""
+
+    homepage = "https://github.com/ORNL/CabanaPD"
+    url = "https://github.com/ORNL/CabanaPD/archive/refs/tags/0.4.0.tar.gz"
+
+    maintainers("streeve")
+    license("BSD 3-Clause", checked_by="cmelone")
+
+    version("0.4.0", sha256="a8971284d3c3d0b5f0bf7ee64e893a2b1e2d094cc0ba62b3c04b1f2fda476bca")
+
+    variant("hdf5", default=False, description="Enable HDF5 support")
+    variant("silo", default=False, description="Enable SILO support")
+    variant("tests", default=False, description="Enable unit tests")
+
+    depends_on("cmake@3.11:", type="build")
+    depends_on("cxx", type="build")
+    depends_on("c", type="build")
+
+    depends_on("cabana+grid@0.7.0:")
+    depends_on("nlohmann-json@3.10.0:")
+    depends_on("cabana+hdf5", when="+hdf5")
+    depends_on("cabana+silo", when="+silo")
+    depends_on("googletest", when="+tests")
+
+    for arch in CudaPackage.cuda_arch_values:
+        cuda_dep = "+cuda cuda_arch={0}".format(arch)
+        depends_on("cabana {0}".format(cuda_dep), when=cuda_dep)
+
+    for arch in ROCmPackage.amdgpu_targets:
+        rocm_dep = "+rocm amdgpu_target={0}".format(arch)
+        depends_on("cabana {0}".format(rocm_dep), when=rocm_dep)
+
+    def cmake_args(self):
+        args = []
+
+        args.append(self.define_from_variant("CabanaPD_ENABLE_TESTING", "tests"))
+
+        if self.spec.satisfies("+rocm"):
+            args.append(self.define("CMAKE_CXX_COMPILER", self.spec["hip"].hipcc))
```

Comment:
https://github.com/spack/spack-packages/blob/9b2fbb2bd89430d5c0c36271e43e2d78397e817d/repos/spack_repo/builtin/packages/beatnik/package.py#L95

Not sure the implications of this, but the comment details why they made a different choice